src/drm/fairplay.js

```diff
@@ -99,9 +99,27 @@ export default class FairPlay extends BaseDrmProtocol {
     FairPlay._logger.debug("License request loaded");
     let request = event.target;
     let keyText = request.responseText.trim();
-    let responseObj = JSON.parse(keyText);
-    let key = FairPlay._base64DecodeUint8Array(responseObj.ckc);
-    FairPlay._keySession.update(key);
+    let responseObj = {};
+    try{
+      responseObj = JSON.parse(keyText);
+    }catch(error){
+      this._licenseRequestFailed();
+    }
+    if (this._validateResponse(responseObj)){
```

Comment:
better have `else` path here and call `this._licenseRequestFailed();` also here.
This way all control flow is done in  `_licenseRequestLoaded` and it is clear when you look at the function that it has multiple error checks, e.g.
```javascript
if (this._validateResponse(responseObj)){
   //do your stuff...
} else {
   this._licenseRequestFailed();
}
```